File: frontend/src/types/vertical.ts
export interface ScopeConfig {
  jurisdictions: string[];
  regulatory_bodies: string[];
  lines_of_business: string[];
  exclusions: string[];
}

export interface CreateVerticalRequest {
  name: string;
  domain_description: string;
  scope: ScopeConfig;
  llm_provider: string;
  expected_source_count_min: number;
  expected_source_count_max: number;
  coverage_target: number;
  rate_limit_ms: number;
  max_concurrent: number;
  timeout_seconds: number;
}

export interface VerticalSummary {
  id: string;
  name: string;
  domain_description: string;
  phase: PipelinePhase;
  source_count: number;
  document_count: number;
  chunk_count: number;
  coverage_score: number;
  created_at: string;
  updated_at: string;
}

export interface PipelinePhaseStatus {
  phase: string;
  status: "pending" | "running" | "complete" | "failed";
  resource_id: string | null;
}

export interface VerticalDetail extends VerticalSummary {
  scope: ScopeConfig;
  llm_provider: string;
  expected_source_count_min: number;
  expected_source_count_max: number;
  coverage_target: number;
  rate_limit_ms: number;
  max_concurrent: number;
  timeout_seconds: number;
  manifest_id: string | null;
  acquisition_id: string | null;
  ingestion_id: string | null;
  last_error: string | null;
  pipeline_status: PipelinePhaseStatus[];
}

export interface VerticalPipelineStatus {
  vertical_id: string;
  phase: PipelinePhase;
  phases: PipelinePhaseStatus[];
  source_count: number;
  document_count: number;
  chunk_count: number;
  coverage_score: number;
}

export interface TriggerResponse {
  vertical_id: string;
  phase: string;
  resource_id: string;
  message: string;
}

export type PipelinePhase =
  | "created"
  | "discovering"
  | "discovered"
  | "acquiring"
  | "acquired"
  | "ingesting"
  | "indexed"
  | "failed";

export const PHASE_LABELS: Record<string, string> = {
  created: "Created",
  discovering: "Discovering",
  discovered: "Discovered",
  acquiring: "Acquiring",
  acquired: "Acquired",
  ingesting: "Ingesting",
  indexed: "Indexed",
  failed: "Failed",
};

export const PHASE_COLORS: Record<string, string> = {
  pending: "var(--text-muted)",
  running: "var(--primary)",
  complete: "var(--success)",
  failed: "var(--danger, #ef4444)",
};

File: backend/app/schemas/vertical.py
"""Pydantic schemas for vertical onboarding and pipeline management."""

from datetime import datetime

from pydantic import BaseModel, Field


class ScopeConfig(BaseModel):
    jurisdictions: list[str] = Field(default_factory=list)
    regulatory_bodies: list[str] = Field(default_factory=list)
    lines_of_business: list[str] = Field(default_factory=list)
    exclusions: list[str] = Field(default_factory=list)


class CreateVerticalRequest(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    domain_description: str = Field(min_length=10)
    scope: ScopeConfig = Field(default_factory=ScopeConfig)
    llm_provider: str = "openai"
    expected_source_count_min: int = Field(default=100, ge=10)
    expected_source_count_max: int = Field(default=300, ge=10)
    coverage_target: float = Field(default=0.85, ge=0.0, le=1.0)
    rate_limit_ms: int = Field(default=2000, ge=100)
    max_concurrent: int = Field(default=5, ge=1, le=20)
    timeout_seconds: int = Field(default=120, ge=10)


class VerticalSummary(BaseModel):
    id: str
    name: str
    domain_description: str
    phase: str
    source_count: int
    document_count: int
    chunk_count: int
    coverage_score: float
    created_at: datetime
    updated_at: datetime


class PipelinePhaseStatus(BaseModel):
    phase: str
    status: str  # pending | running | complete | failed
    resource_id: str | None = None


class VerticalDetail(BaseModel):
    id: str
    name: str
    domain_description: str
    scope: ScopeConfig
    llm_provider: str
    expected_source_count_min: int
    expected_source_count_max: int
    coverage_target: float
    rate_limit_ms: int
    max_concurrent: int
    timeout_seconds: int
    phase: str
    manifest_id: str | None
    acquisition_id: str | None
    ingestion_id: str | None
    source_count: int
    document_count: int
    chunk_count: int
    coverage_score: float
    last_error: str | None
    created_at: datetime
    updated_at: datetime
    pipeline_status: list[PipelinePhaseStatus] = []


class VerticalPipelineStatus(BaseModel):
    vertical_id: str
    phase: str
    phases: list[PipelinePhaseStatus]
    source_count: int
    document_count: int
    chunk_count: int
    coverage_score: float


class TriggerResponse(BaseModel):
    vertical_id: str
    phase: str
    resource_id: str
    message: str
